src/printer.js

```diff
@@ -1955,6 +1955,26 @@ function shouldGroupLastArg(args) {
     (!penultimateArg || penultimateArg.type !== lastArg.type);
 }
 
+function shouldGroupFirstArg(args) {
+  if (args.length !== 2) {
+    return false;
+  }
+
+  const firstArg = args[0];
+  const secondArg = args[1];
+  return (firstArg.type === 'FunctionExpression' ||
+    (firstArg.type === 'ArrowFunctionExpression' &&
+      firstArg.body.type === 'BlockStatement')) &&
+      // Only allow one simple trailing arg.
```

Comment:
I think that this is going to be very confusing.

`timeout` is good but `this.timeout`, `1000 * timeout`, `this.getTimeout()`, `this.props.timeout * 1000` is not good.

I'm not really sure what the best solution is but I figured I would give you some ways people are doing it from the fb codebase.